l2geth/rollup/sync_service.go

```diff
@@ -869,9 +883,24 @@ func (s *SyncService) applyTransactionToTip(tx *types.Transaction) error {
 		s.SetLatestEnqueueIndex(queueIndex)
 	}
 
+	// This reset function MUST be called on any error case
+	reset := func() {
+		s.SetLatestL1Timestamp(ts)
+		s.SetLatestL1BlockNumber(bn)
+		s.SetLatestIndex(index)
+		s.SetLatestEnqueueIndex(queueIndex)
```

Comment:
Per @cfromknecht's [comment](https://github.com/ethereum-optimism/optimism/pull/2395#discussion_r841995467) should we reset the enqueue index?
cc @tynes 

Did we have tests for the deadlock bug?